harness/determined/keras/_tf_keras_trial.py

```diff
@@ -500,24 +539,52 @@ def compute_validation_metrics(self) -> workload.Response:
             validation_steps,
         ) = self._validation_input_manager.get_validation_input_and_num_batches()
 
-        metrics_values = self.model.evaluate(validation_data, steps=validation_steps, verbose=0)
+        evaluate_kwargs = {
+            "steps": validation_steps,
+            "verbose": 0,
+        }
+
+        # Starting in TF 2.2 users may define custom test_step() that do
+        # not use the model metrics.
+        use_model_metrics = version.parse(tf.__version__) < version.parse("2.2.0")
+        if not use_model_metrics:
+            evaluate_kwargs["return_dict"] = True
+
+        metrics_values = self.model.evaluate(validation_data, **evaluate_kwargs)
+        logging.debug(f"Worker finished model.evaluate() with metrics: {metrics_values}.")
 
         # If the model was compiled with metrics=None, metrics_value will be a single value.
-        if not isinstance(metrics_values, (tuple, list)):
+        if not isinstance(metrics_values, (tuple, list, dict)):
             metrics_values = (metrics_values,)
 
         if self.hvd_config.use:
-            for index, metric_value in enumerate(metrics_values):
-                metrics_values[index] = np.array(hvd.allreduce(metric_value))
+            if isinstance(metrics_values, dict):
+                for metric_name in metrics_values.keys():
```

Comment:
same as you said on mine: is this guaranteed to iterate the same on every worker?